lib/charms/loki_k8s/v0/loki_push_api.py

```diff
@@ -258,15 +360,42 @@ def __init__(self, message="Alert rule is not well formed"):
 class LokiProvider(RelationManagerBase):
     """A LokiProvider class."""
 
-    def __init__(self, charm, relation_name: str):
+    def __init__(self, charm, relation_name: Optional[str] = None):
```

Comment:
Better to add the default relation name as argument default and have the argument type of relation_name to be `str`, easier to understand